### PR TITLE
Merge branch binary_extractor  into branch datetime_extractor

### PR DESCRIPTION
**Pull request by lschiesser**

### CODE REVIEW

File: Documentation.md [hunk 1]
```diff
@@ -1,5 +1,17 @@
 ## Feature Extraction
 
+### Binary Feature Extraction
+
+**Motivation**
+
+There are some aspects of a tweet where it is not necessary to know how much of that aspect is in the tweet or what specific content the aspect has.
+Instead, it is just interesting to know if the aspect is there or not. 
+Therefore, we decided to implement a class that produces binary features from the dataset.
```
Comment (polina-tsvilodub) [inline Documentation.md]:
```suggestion
Therefore, we decided to implement a class that produces binary features from the dataset, indicating the presence (or absense) of a given aspect.
```

File: Documentation.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+
+**Implementation**
+The binary feature eextractor is implemented in the class `BinaryFeatureExtractor` which is located in `code/feature_extraction/binary_features.py`.
```
Comment (polina-tsvilodub) [inline Documentation.md]:
```suggestion
The binary feature eextractor is implemented in the class `BinaryFeatureExtractor` which is located in `code/feature_extraction/binary_features.py` and is a subclass of `FeatureExtractor`.
```

File: code/feature_extraction/binary_features.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+class BinaryFeatureExtractor(FeatureExtractor):
+    
+    def __init__(self, input_columns, output_column):
+        self.input_columns = input_columns
+        super().__init__([input_columns], output_column)
+    
+    def _get_values(self, inputs):
+        result = []
+        # only process feature if it's a list
+        # one feature is combined from two columns in the dataset called media present
```
Comment (polina-tsvilodub) [inline code/feature_extraction/binary_features.py]:
```suggestion
        # one feature called media present is combined from two columns in the dataset 
```

File: code/feature_extraction/binary_features.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        # it is combined from the columns photos and video
+        if "photos" in self.input_columns:
+            # column photos comes as a list of json objects
+            # column photos comes as a boolean number indicating presence (1) or absence (0) of video
```
Comment (polina-tsvilodub) [inline code/feature_extraction/binary_features.py]:
```suggestion
            # column video comes as a boolean number indicating presence (1) or absence (0) of video
```

File: code/feature_extraction/binary_features.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            for tmp in inputs[0]:
+                # this expression returns 1 if there is media persent (if list of photos bigger than 0
+                # and/or if there video == 1) and 0 if media is absent (list of photos == 0 and video == 0)
```
Comment (polina-tsvilodub) [inline code/feature_extraction/binary_features.py]:
```suggestion
                # and/or if video == 1) and 0 if media is absent (list of photos == 0 and video == 0)
```

File: code/feature_extraction/binary_features.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+                result.append(int(len(tmp[0]) > 0 or bool(tmp[1])))
```
Comment (polina-tsvilodub) [inline code/feature_extraction/binary_features.py:29]: Is tmp really a line containing both the photo and the video? I feel like it only gets the photo json entry...

Comment (polina-tsvilodub) [reply code/feature_extraction/binary_features.py:29]: ohh i think it's my mistakes because the input looks like [["video", "photo"]] correct?

**Review by polina-tsvilodub — APPROVED** [review]
Binary feature extractor looks good, added some cosmetic comments.

File: code/feature_extraction/binary_features.py (hunk 6 above)
Comment (lschiesser) [reply code/feature_extraction/binary_features.py:29]: Correct. It does contain two columns

**Pull request merged**